src/yarp.c

```diff
@@ -1484,6 +1486,9 @@ missing(yp_token_t *token, const char *location) {
 static yp_node_t *
 parse_expression(yp_parser_t *parser, binding_power_t binding_power, const char *message);
 
+static yp_node_t *
+parse_left_hand_side(yp_parser_t *parser, binding_power_t binding_power, const char *message);
```

Comment:
C was having a hard time finding our `parse_left_hand_side` method, so we thought to follow the pattern of `parse_expression` and include this declaration here. I'm not really sure why we need this!

Reply:
So C is a one-pass compiler, meaning it has to know about all of the symbols in the program when they are referenced. This is called a forward declaration, that tells the compiler "hey I'm going to have a function later that is named like this and has these types". It allows the compiler to keep compiling, even without the function body. When it finds it lower in the file, it'll fill in the body.

In this case though, you should move the function up to this point, because it's not necessary that it lives lower. In that case you will be able to remove the forward declaration since it all can live right here.

Reply:
Ah thanks for the explanation! Moved the method up here and removed the declaration.